packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/ExoPlayerEventListener.java

```diff
@@ -49,23 +53,59 @@ private void sendInitialized() {
     int width = videoSize.width;
     int height = videoSize.height;
     if (width != 0 && height != 0) {
-      int rotationDegrees = videoSize.unappliedRotationDegrees;
-      // Switch the width/height if video was taken in portrait mode
-      if (rotationDegrees == 90 || rotationDegrees == 270) {
+      int reportedRotationCorrection = 0;
+
+      if (Build.VERSION.SDK_INT <= 21) {
+        // On API 21 and below, Exoplayer may not internally handle rotation correction
+        // and reports it through VideoSize.unappliedRotationDegrees. We may apply it to
+        // fix the case of upside-down playback.
+        reportedRotationCorrection = videoSize.unappliedRotationDegrees;
+        rotationCorrection = getRotationCorrectionFromUnappliedRotation(reportedRotationCorrection);
+      } else if (Build.VERSION.SDK_INT < 29) {
```

Comment:
Should this use the new api matan added `handlesCropAndRotation` instead? https://github.com/flutter/engine/pull/55434/files

Reply:
It's available in the _next_ stable unfortunately, i.e. the one to be published (but not yet).

Reply:
@camsim99 consider a TODO here to use the new api.